lib/msal-browser/src/controllers/StandardController.ts

```diff
@@ -426,35 +449,50 @@ export class StandardController implements IController {
                                     "handleRedirectResponse returned result, acquire token success"
                                 );
                             }
+                            rootMeasurement.end({ success: true });
                         }
                         this.eventHandler.emitEvent(
                             EventType.HANDLE_REDIRECT_END,
                             InteractionType.Redirect
                         );
+                        rootMeasurement.end({ success: false });
 
                         return result;
                     })
                     .catch((e) => {
+                        const eventError = e as EventError;
                         // Emit login event if there is an account
                         if (loggedInAccounts.length > 0) {
                             this.eventHandler.emitEvent(
                                 EventType.ACQUIRE_TOKEN_FAILURE,
                                 InteractionType.Redirect,
                                 null,
-                                e as EventError
+                                eventError
                             );
                         } else {
                             this.eventHandler.emitEvent(
                                 EventType.LOGIN_FAILURE,
                                 InteractionType.Redirect,
                                 null,
-                                e as EventError
+                                eventError
                             );
                         }
                         this.eventHandler.emitEvent(
                             EventType.HANDLE_REDIRECT_END,
                             InteractionType.Redirect
                         );
+                        if (eventError instanceof AuthError) {
+                            rootMeasurement.end({
+                                success: false,
+                                errorCode: eventError.errorCode,
+                                subErrorCode: eventError.subError,
+                            });
+                        } else {
+                            rootMeasurement.end({
+                                success: false,
+                                errorCode: eventError?.message,
```

Comment:
I've found previously that tracking arbitrary error messages like this often results in long, unique strings and may even contain PII (think JSON.parse error for example that may leak the string that failed parsing)

Reply:
That's a good point, thanks. Removed it.